src/piton/etl_pipelines/omop.py

```diff
@@ -0,0 +1,150 @@
+"""An ETL script for doing an end to end transform of generic OMOP data into a PatientDatabase."""
+
+import argparse
+import datetime
+import json
+import logging
+import os
+import resource
+from typing import Callable, Dict, Optional, Sequence
+
+from piton import Patient
+from piton.datasets import EventCollection, PatientCollection
+from piton.extractors.csv import run_csv_extractors
+from piton.extractors.omop import get_omop_csv_extractors
+from piton.transforms import delta_encode, remove_nones, remove_short_patients
+
+
+def _get_generic_omop_transformations() -> Sequence[Callable[[Patient], Optional[Patient]]]:
+    """Get the list of current OMOP transformations."""
+    # All of these transformations are information preserving
+    transforms: Sequence[Callable[[Patient], Optional[Patient]]] = [
+        remove_nones,
+        delta_encode,
+        remove_short_patients,
+    ]
+
+    return transforms
+
+
+def etl_generic_omop_program() -> None:
+    """Extract data from an generic OMOP source to create a piton PatientDatabase."""
+    parser = argparse.ArgumentParser(description="An extraction tool for generic OMOP sources")
+
+    parser.add_argument(
+        "omop_source",
+        type=str,
+        help="Path of the folder to the omop source",
+    )
+
+    parser.add_argument(
+        "target_location",
+        type=str,
+        help="The place to store the extract",
+    )
+
+    parser.add_argument(
+        "temp_location",
+        type=str,
+        help="The place to store temporary files",
+        default=None,
+    )
+
+    parser.add_argument(
+        "--num_threads",
+        type=int,
+        help="The number of threads to use",
+        default=1,
+    )
+
+    args = parser.parse_args()
+
+    soft, hard = resource.getrlimit(resource.RLIMIT_NOFILE)
```

Comment:
Again, some code comments would go a long way. What are these doing? Why do we set them?